translation agile_software_development and devops (#356)
cncf/glossary: content/it/devops.md

---
title: DevOps
status: Completed
category: concept
---

## Cos'è

> DevOps è una metodologia in cui i team sono responsabili dell'intero processo dallo sviluppo delle applicazioni fino alle attività di installazione, configurazione e manutenzione in produzione, da cui DevOps. DevOps va oltre l'implementazione di una serie di tecnologie e richiede un cambiamento completo nella cultura e nei processi. DevOps richiede gruppi di ingegneri che lavorano su piccoli componenti (invece di un'intera feature), diminuendo i passaggi di mano - una fonte comune di errori. 

Reviewer: ```suggestion
DevOps è una metodologia in cui i team sono responsabili dell'intero processo: dallo sviluppo delle applicazioni fino alle attività di installazione, configurazione e manutenzione in produzione, da cui DevOps. Il concetto va oltre l'implementazione di una serie di tecnologie e richiede un cambiamento completo nella cultura e nei processi. DevOps richiede gruppi di ingegneri che lavorano su piccoli componenti (invece di un'intera feature), diminuendo i passaggi di mano - una fonte comune di errori. 
```

## Quale problema affronta

> Tradizionalmente, nelle organizzazioni complesse con [tightly-coupled](/tightly_coupled_architectures/) [monolithic apps](/monolithic_apps/), il lavoro era generalmente frammentato tra più team. Questo portava a numerosi passaggi di mano e a lunghi tempi di consegna. Ogni volta che un componente o un aggiornamento era pronto, veniva messo in una coda per il team successivo. Poiché gli individui lavoravano solo su una piccola parte del progetto, questo approccio portava ad una mancanza di proprietà sull'intero progetto. Il loro obiettivo era quello di passare il lavoro al gruppo successivo e non di consegnare la giusta funzionalità al cliente - un evidente e chiaro disallineamento delle priorità.

Reviewer 2: - tightly-coupled monolithic apps > applicazioni monolitiche dai componenti strettamente accoppiati
- in una coda > in coda
- una mancanza di proprietà > una mancanza di titolarità (o visione o gestione)

> Tradizionalmente, nelle organizzazioni complesse con [applicazioni monolitiche dai componenti strettamente accoppiati](/tightly_coupled_architectures/) [monolithic apps](/monolithic_apps/), il lavoro era generalmente frammentato tra più team. Questo portava a numerosi passaggi di mano e a lunghi tempi di consegna. Ogni volta che un componente o un aggiornamento era pronto, veniva messo in coda per il team successivo. Poiché gli individui lavoravano solo su una piccola parte del progetto, questo approccio portava ad una mancanza di titolarità (ownership) sull'intero progetto. Il loro obiettivo era quello di passare il lavoro al gruppo successivo e non di consegnare la giusta funzionalità al cliente - un evidente e chiaro disallineamento delle priorità.

Reviewer: per conservare l'approccio di spiegare tra parentesi termini che giudichiamo più completi in inglese
```suggestion
Tradizionalmente, nelle organizzazioni complesse con [applicazioni monolitiche](/monolithic_apps/) dai [componenti strettamente accoppiati](/tightly_coupled_architectures/), il lavoro era generalmente frammentato tra più team. Questo portava a numerosi passaggi di mano e a lunghi tempi di consegna. Ogni volta che un componente o un aggiornamento era pronto, veniva messo in coda per il team successivo. Poiché gli individui lavoravano solo su una piccola parte del progetto, questo approccio portava ad una mancanza di ownership (titolarità) sull'intero progetto. Il loro obiettivo era quello di passare il lavoro al gruppo successivo e non di consegnare la giusta funzionalità al cliente - un evidente e chiaro disallineamento delle priorità.
```

> Quando il codice arrivava finalmente in produzione, passava attraverso così tanti sviluppatori ed in attesa in così tante code che era difficile rintracciare l'origine del problema se il codice non funzionava. DevOps capovolge questo approccio.

Reviewer 2: - passava attraverso così tanti sviluppatori ed in attesa in così tante code che era difficile 
> passava da così tanti sviluppatori, stando in attesa in così tante code, che diventava difficile

> Quando il codice arrivava finalmente in produzione, passava da così tanti sviluppatori, stando in attesa in così tante code, che diventava difficile rintracciare l'origine del problema se il codice non funzionava. DevOps capovolge questo approccio.

Reviewer: Questo è solo un gusto personale :D
```suggestion
Quando il codice arrivava finalmente in produzione, passava da così tanti sviluppatori, stando in attesa in così tante code, che diventava difficile rintracciare l'origine del problema se il codice non funzionava. DevOps inverte questo approccio.
```

## In che modo aiuta

> Avere un team che è responsabile dell'intero ciclo di vita di un'applicazione porta a minimizzare i passaggi di mano, a ridurre il rischio quando si installa in produzione, a migliorare la qualità del codice perché i team sono anche responsabili di come il codice si comporta in produzione e ad aumentare la soddisfazione dei dipendenti grazie alla maggiore autonomia e proprietà.

Reviewer 2: anche in questo caso _ownership_ lo lascerei invariato, oppure lo tradurre con titolarità di progetto

Author: certo, per ora meglio ownership, titolarità non credo che sia proprio adeguato, magari cerca un altro sinonimo

> Avere un team che è responsabile dell'intero ciclo di vita di un'applicazione porta a minimizzare i passaggi di mano, a ridurre il rischio quando si installa in produzione, a migliorare la qualità del codice perché i team sono anche responsabili di come il codice si comporta in produzione e ad aumentare la soddisfazione dei dipendenti grazie alla maggiore autonomia e proprietà (ownership).

Reviewer: tra le altre: `ownership` è un concetto già introdotto nel paragrafo precedente
```suggestion
Avere un team che sia responsabile dell'intero ciclo di vita di un'applicazione, porta a minimizzare i passaggi di mano, a ridurre i rischi di installazione in produzione, a migliorare la qualità del codice (perché i team sono anche responsabili di come il codice "si comporta" in produzione) e ad aumentare la soddisfazione dei dipendenti grazie alla maggiore autonomia e ownership.
```

## Quali problematiche affronta
Tradizionalmente, nelle organizzazioni complesse con [applicazioni monolitiche dai componenti strettamente accoppiati](/tightly_coupled_architectures/) [monolithic apps](/monolithic_apps/), il lavoro era generalmente frammentato tra più team. Questo portava a numerosi passaggi di mano e a lunghi tempi di consegna. Ogni volta che un componente o un aggiornamento era pronto, veniva messo in coda per il team successivo. Poiché gli individui lavoravano solo su una piccola parte del progetto, questo approccio portava ad una mancanza di titolarità (ownership) sull'intero progetto. Il loro obiettivo era quello di passare il lavoro al gruppo successivo e non di consegnare la giusta funzionalità al cliente - un evidente e chiaro disallineamento delle priorità.

Quando il codice arrivava finalmente in produzione, passava da così tanti sviluppatori, stando in attesa in così tante code, che diventava difficile rintracciare l'origine del problema se il codice non funzionava. DevOps capovolge questo approccio.

## In che modo aiuta

> Avere un team che è responsabile dell'intero ciclo di vita di un'applicazione porta a minimizzare i passaggi di mano, a ridurre il rischio quando si installa in produzione, a migliorare la qualità del codice perché i team sono anche responsabili di come il codice si comporta in produzione e ad aumentare la soddisfazione dei dipendenti grazie alla maggiore autonomia e proprietà (ownership).

Reviewer 2: alla maggiore autonomia e proprietà (ownership) --> titolarità di progetto (ownership) oppure responsabilità (ownership)

Author: ok
